### PR TITLE
Wip/hughsie/fu volume

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: libfwupdplugin/fu-common.c [hunk 1]
```diff
@@ -2069,3 +2070,196 @@ fu_common_is_cpu_intel (void)
...
+ * @kind: A volume kind, typically a GUID
+ * @error: A #GError or NULL
+ *
+ * Call into the plugin's get results routine
+ *
+ * Finds all volumes of a specific type
+ *
+ * Returns: (transfer container) (element-type FuVolume): a #GPtrArray, or %NULL if the kind was not found
+ *
+ * Since: 1.5.0
```
Comment (superm1) [inline libfwupdplugin/fu-common.c:2128]: Not going to be a backport to 1.4.6 feature?

Comment (hughsie) [reply libfwupdplugin/fu-common.c:2128]: I was tempted to push 1.5.0 soon. I can do these for 1.4.6 if you'd prefer.

File: libfwupdplugin/fu-common.c [hunk 2]
```diff
@@ -2069,3 +2070,196 @@ fu_common_is_cpu_intel (void)
 #endif
 	return FALSE;
 }
+
+#define UDISKS_DBUS_SERVICE		"org.freedesktop.UDisks2"
+#define UDISKS_DBUS_PATH		"/org/freedesktop/UDisks2/Manager"
+#define UDISKS_DBUS_MANAGER_INTERFACE	"org.freedesktop.UDisks2.Manager"
+#define UDISKS_DBUS_PART_INTERFACE 	"org.freedesktop.UDisks2.Partition"
+#define UDISKS_DBUS_FILE_INTERFACE	"org.freedesktop.UDisks2.Filesystem"
```
[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-common.c:2078]: sure can; i wanted your sanity check before I spent too much time on the branch

File: libfwupdplugin/fu-common.c (hunk 1 above)
Comment (superm1) [reply libfwupdplugin/fu-common.c:2128]: Well I'm just hesitant on 1.5.0 until there is more !intel in HSI..

File: libfwupdplugin/fu-common.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (superm1) [reply libfwupdplugin/fu-common.c:2078]: rest of it _looks_ good (didn't actually test locally though).

**Comment by superm1** [comment]
Branch looks good, the only thing I would think is possibly missing is self tests for the volume stuff (although I guess you can argue that uefi is exercising a lot of it)

**Comment by hughsie** [comment]
> self tests for the volume stuff

Do you know how to mock that up? It's all using UDisks rather than poking at mtab now.

**Comment by superm1** [comment]
> Do you know how to mock that up? It's all using UDisks rather than poking at mtab now.

Oh True.  I was going to suggest umockdev, but that might not actually work with udisks.  Considering this, maybe skipping tests is the right answer for this stuff.

File: libfwupdplugin/fu-common.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-common.c]: Fair point. Moved to 1.4.6

File: libfwupdplugin/fu-common.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-common.c]: Fixed.

**Comment by hughsie** [comment]
I'm particularly proud of `fu_volume_locker()` as we didn't do a terribly good job of unmounting the ESP on the failure path before.

**Pull request merged**